app/js/stores/AnswerStore.js

```diff
@@ -1,114 +1,88 @@
-
-    case ZhishiConstants.QUESTION_UPDATE:
-      if (action.data) {
-        loadAnswers(action.data.id, action.data.answers)
-        AnswerStore.emitChange();
-      }
-      break;
-
-    case ZhishiConstants.ANSWER_INDEX:
-      if (action.data) {
-        loadAnswers(action.data.question_id, action.data.answers)
-        AnswerStore.emitChange();
-      }
-      break;
-
-    // case ZhishiConstants.QUESTION_UPDATE:
-    //   if (action.data && action.data.answers) {
-    //     loadAnswers(action.data.id, action.data.answers)
-    //   }
-    //   break;
-
-    case ZhishiConstants.ANSWER_EDIT:
-      if (action.data) {
-        edit(action.data.question_id, action.data.id)
-        AnswerStore.emitChange();
-      }
-      break;
 
-    case ZhishiConstants.ANSWER_UPDATE:
-      if (action.data) {
-        update(action.data);
-        AnswerStore.emitChange();
-      }
-      break;
+  edit(data) {
+    this._answers[data.question_id][data.id]['status'] = 'editing editor-content';
+  }
 
-    case ZhishiConstants.ANSWER_UPDATE_VOTES:
-      if (action.data) {
-        update_votes_count({...action.data})
-        AnswerStore.emitChange();
-      }
-      break;
+  update(answer) {
+    Common.update(this._answers[answer.question_id], answer.id, answer);
+  }
+  getAnswer(question_id, id) {
+    return this._answers[question_id][id];
+  }
 
-    default:
-      // nothing for now
+  getAnswers(question_id) {
+    return this._answers[question_id];
   }
-});
+  getTopAnswers() {
+    return this._top_answers;
+  }
+  update_votes_count(data) {
+    this._answers[data.meta.question_id][data.id]['votes_count'] = data.votes_count.response;
+    this._answers[data.meta.question_id][data.id]['user_vote'] = data.value;
+  }
+  _registerActions(action) {
+    switch (action.actionType) {
+
+      case ZhishiConstants.QUESTION_UPDATE:
+        if (action.data) {
+          this.loadAnswers(action.data.id, action.data.answers);
+          this.emitChange();
+        }
+        break;
+
+      case ZhishiConstants.ANSWER_INDEX:
+        if (action.data) {
+          this.loadAnswers(action.data.question_id, action.data.answers);
+          this.emitChange();
+        }
+        break;
+
+      // case ZhishiConstants.QUESTION_UPDATE:
+      //   if (action.data && action.data.answers) {
+      //     loadAnswers(action.data.id, action.data.answers)
+      //   }
+      //   break;
+
+      case ZhishiConstants.ANSWER_EDIT:
+        if (action.data) {
+          this.edit(action.data);
+          this.emitChange();
+        }
+        break;
+
+      case ZhishiConstants.ANSWER_UPDATE:
+        if (action.data) {
+          this.update(action.data);
+          this.emitChange();
+        }
+        break;
+
+      case ZhishiConstants.ANSWER_UPDATE_VOTES:
+        if (action.data) {
+          this.update_votes_count(action.data);
+          this.emitChange();
+        }
+        break;
+
+      default:
+        // nothing for now
+    }
+  }
+}
```

Comment:
So private methods isn't implemented in es6 yet. But it's sill better than having global functions. That way the function can only be called by the instance of the class. If we are to be using typescript as our javascript transpiler it would have made this easy. The idea of BaseStore is just to avoid some basic repitations as with all stores sharing the same common properties and methods.

More so this was basically to avoid breaking changes in the app. We could have used
```javascript
get methodName() {
return 'method name'
}
```
for the functions we only need returned variables from. this makes it look like accessing the instance properties